### ISSUE TITLE
Track1/6_SPD_EST issues

### ISSUE THREAD

**Issue by JoyceLiang-sudo** [zhengthomastang/2018AICity_TeamUW#21]
Thank you very much for sharing. I want to ask if speed detection is implemented in Python?

**Comment by zhengthomastang** [comment]
Sorry. Most of our implementations are in C++ and Matlab. But for our single-camera tracking part, there is a new version using PyTorch. You can also choose to use Python-based object detectors.

**Comment by JoyceLiang-sudo** [comment]
> Sorry. Most of our implementations are in C++ and Matlab. But for our single-camera tracking part, there is a new version using PyTorch. You can also choose to use Python-based object detectors.

For the speed estimation,would you like to tell me how to get the camera parameters? Is it based on the camera that shot this video?

**Comment by zhengthomastang** [comment]
@tabu666 Yes. The camera parameters are different for each camera. You can use our provided camera calibration tool to compute them: https://github.com/zhengthomastang/2018AICity_TeamUW/tree/master/Track1/2_CAM_CAL

**Comment by JoyceLiang-sudo** [comment]
@zhengthomastang OK,thanks for your help!

**Comment by bitasd** [comment]
@zhengthomastang  Hi Thomas!
Could you please let me know:
1.why the speed has been scaled by 1.25, 1.05, 0.8, ets depending on the video (line 151 in _main.cpp_), and how these numbers have been drawn?
2. why the accumulated distance between frames has been multiplied by **2.23694**  (line 348 in _main.cpp_)
3. why the number of used frames for the computation of average speed is set to 15 for some videos and 31 in some others
Your response would help me out a lot!
Thanks!!!

**Comment by zhengthomastang** [comment]
@bitasd Thanks for the questions. For (1) and (2), the scales on the speed values have been fine-tuned based on the actual speed limits and the evaluated performance on the test set. For (3), it is because the frame rates for the videos are different. We also considered the viewing perspectives to determine what a reliable range of frames for speed estimation should be.

**Comment by zhengthomastang** [comment]
@bitasd The constant **2.23694**  (line 348 in main.cpp) is for converting meters/second to miles/hour. Only the parameters in (1) were tuned.

**Comment by bitasd** [comment]
> You can also choose to use Python-based object detectors.
> , there is a new version using PyTorch
> But for our single-camera tracking part, there is a new version using PyTorch

Could you please point us to this python implementation? greatly appreciate it.

**Comment by zhengthomastang** [comment]
Hi. You can check out Faster R-CNN, SSD, YOLO, etc.
